rmw_zenoh_cpp/src/detail/rmw_data_types.hpp

```diff
@@ -131,4 +129,66 @@ struct rmw_subscription_data_t
   std::condition_variable * condition{nullptr};
 };
 
+
+///==============================================================================
+
+// z_owned_closure_query_t
+void service_data_handler(const z_query_t * query, void * service_data);
+
+void client_data_handler(z_owned_reply_t * reply, void * client_data);
+
+
+///==============================================================================
+
+struct rmw_service_data_t
+{
+  unsigned int get_new_uid();
+
+  const char * keyexpr;
+  z_owned_queryable_t qable;
+
+  const void * request_type_support_impl;
+  const void * response_type_support_impl;
+  const char * typesupport_identifier;
+  RequestTypeSupport * request_type_support;
+  ResponseTypeSupport * response_type_support;
+
+  rmw_context_t * context;
+
+  // Map to store the query id and the query.
+  // The query handler is saved as it is needed to answer the query later on.
+  std::unordered_map<unsigned int, std::unique_ptr<z_owned_query_t>> id_query_map;
```

Comment:
Another suggestion to consider: It might be simpler to store the owned query directly. Once we're done "taking" the reply, we can just drop it without needing to reset the unique ptr AND dropping the value it was managing. 

```suggestion
  std::unordered_map<unsigned int, z_owned_query_t> id_query_map;
```